WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -68,6 +86,22 @@ class PostListMainViewModel @Inject constructor(
         _postListAction.postValue(PostListAction.NewPost(site))
     }
 
+    fun updateAuthorFilterSelection(position: Int) {
```

Comment:
I think I am missing something very obvious here, but why does this method have a `position` parameter instead of the `AuthorFilterSelection`? In the `start` method we are converting the `enum` to `ordinal` and then converting it back to the same `enum`, what am I missing?

Reply:
I was missing something obvious, this is accepting an `Int` because of the `OnItemSelectedListener` in the activity. I think this is all the more reason to separate the internal and external usages of `updateAuthorFilterSelection` because it feels really bad to convert an enum to an `ordinal` just to convert it back. I think the public facing method should update the `AppPrefs` and convert the position to the `enum` and let the internal method handle the rest.

Reply:
I think the various changes have resolved this issue.